### PR TITLE
Add clear start-to-finish instructions for running the application locally

### PR DESCRIPTION
**Pull request by mpidcock**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,126 @@
...
+   * [application-test.properties](src/main/resources/application-test.properties) sets the datasource url to `messaging_test`, in addition to other important settings
+   * Currently, this project does not have version controlled run configurations
+   * We suggest altering all [run configuration templates](https://www.jetbrains.com/help/idea/run-debug-configuration.html#templates) to use the `test` profile so that they will pick up config changes automatically
+5. Run the `MessagingApplication` locally
+   * You will also have to set up this run configuration yourself
+   * Set the spring profile to 'test'
+   * Override the database url by setting `SPRING_DATASOURCE_URL=jdbc:postgresql://localhost:5432/messaging`
+   * Any service providers you wish to integrate with will require the API keys to be set using accounts you have access to
+     * Mailgun integration requires:
+       * `MAILGUN_API_KEY`
```
Comment (vrajmohan) [inline README.md:62]: We also need `MAILGUN_API_DOMAIN` and `MAILGUN_API_FROM`.

File: README.md [hunk 2]
```diff
@@ -1,43 +1,126 @@
...
+      * Test database = `jdbc:postgresql://localhost:5432/messaging_test`
+    * To run the application locally, you will need to create a second database
+      * Local database = `jdbc:postgresql://localhost:5432/messaging`
+    * The database in the deployed environment (i.e. Staging) can be tunneled into using Aptible credentials
+4. Run all tests
+   * Ensure you are using the `test` [spring profile](https://docs.spring.io/spring-boot/docs/1.2.0.M1/reference/html/boot-features-profiles.html)
+   * [application-test.properties](src/main/resources/application-test.properties) sets the datasource url to `messaging_test`, in addition to other important settings
+   * Currently, this project does not have version controlled run configurations
+   * We suggest altering all [run configuration templates](https://www.jetbrains.com/help/idea/run-debug-configuration.html#templates) to use the `test` profile so that they will pick up config changes automatically
+5. Run the `MessagingApplication` locally
```
[the remark this answers is not in the document]
Comment (mpidcock) [reply README.md:56]: Should I rename [MessagingApplication.java](https://github.com/codeforamerica/messaging/blob/main/src/main/java/org/codeforamerica/messaging/MessagingApplication.java)? I can at least add `.java` to be clearer in the instuctions.

**Comment by mpidcock** [comment]
> 1. We may be better off keeping README.md small and linking an INSTALLING.md file.

Agree, separated it

> 2. We need some instructions/examples on creating `handlebars` templates,

I started working on a Google doc for more in-depth instructions, but it isn't geared toward developers. More like things to think about and options available for template creation. For developers I'm not sure they need anything other than something simple to test with.

> 3. We should have a few sentences on actual usage, viz., sending a single message, sending bulk messages, scheduling messages, etc and leverage the Swagger docs.

I agree, but thought this might be out of scope for this doc. I do mention the swagger docs and push the reader to consult those since they will still be evolving for a bit. We can make a story to improve the generated docs to be more informative and give better examples.

**Review by vrajmohan — APPROVED** [review]
Looks good.

**Pull request merged**